app/services/activitypub/process_status_update_service.rb

```diff
@@ -17,16 +17,19 @@ def call(status, json)
 
     last_edit_date = status.edited_at.presence || status.created_at
 
+    # To avoid any changes prior to this service affecting it
+    @status.clear_changes_information
```

Comment:
I understand why you'd do that but I feel somewhat uncomfortable about that… is there any case in which `@status` is dirty when passed to `ActivityPub::ProcessStatusUpdateService`? If so, why and what should happen? If not, wouldn't it better to raise an exception if `@status` is dirty?

Reply:
This happens in tests, in production there is no place where a status would be modified between being loaded and being passed into this class, however there *should not* ever be such a case where it's necessary so I think it's safe to have this reset.

Reply:
Ok… what about raising an issue if `@status.has_changes_to_save?` then calling `clear_changes_information` to clear the previous changes bit, and if any test fails, ensure that the status is saved before being passed?

Reply:
`has_changes_to_save?` is about unsaved changes (and is also alias of `changed?` which we more commonly use), and unsaved changes have never been an issue here or in tests, it's only about `previous_changes` for which there is no one-liner check. I see no downside to just doing the `clear_changes_information` straight away to clear `previous_changes`.

Reply:
I know it hasn't been an issue, I'm just saying it doesn't cost much to *ensure* this assumption is met in the future by adding a check.

i.e., doing:
```ruby
raise ArgumentError if status.changed?
status.clear_changes_information
```